analytic_engine/src/sst/parquet/async_reader.rs

```diff
@@ -51,12 +56,67 @@ use crate::sst::{
 
 type SendableRecordBatchStream = Pin<Box<dyn Stream<Item = Result<ArrowRecordBatch>> + Send>>;
 
+pub type AsyncFileReaderRef = Arc<dyn AsyncFileReader>;
+
+#[async_trait]
+pub trait AsyncFileReader: Send + Sync {
```

Comment:
This trait seems unnecessary, I can't see why this is required.

Reply:
Actually, it is. This is an alternative for `ObjectStore` in the async_reader module.